core/src/com/unciv/ui/popups/CityScreenConstructionMenu.kt

```diff
@@ -100,6 +110,20 @@ class CityScreenConstructionMenu(
             it.addToQueue(construction, true)
     }
 
-    private fun canRemoveAllQueues() = anyCity { it.isBeingConstructedOrEnqueued(constructionName) }
+    private fun canRemoveAllQueues() = allCitiesEntryValid { it.isBeingConstructedOrEnqueued(constructionName) }
     private fun removeAllQueues() = forAllCities { it.removeAllByName(constructionName) }
+
+    private fun canDisable() = constructionName !in dontAutoAssignConstructions &&
+        constructionName != PerpetualConstruction.idle.name
+    private fun disableEntry() {
+        dontAutoAssignConstructions.add(constructionName)
+        settings.save()
+    }
+
+    private fun canEnable() = constructionName in dontAutoAssignConstructions &&
+        constructionName != PerpetualConstruction.idle.name
```

Comment:
We have the construction itself, we can compare it to idle directly, not via names
Also idle should never be in the disabled constructions, and if somehow it IS then we want to be able to enable it, this check seems unnecessary